pkg/getty/README.md

```diff
@@ -10,7 +10,7 @@ If you want a console getty, add the following to your `moby.yml`:
 ```
 services:
   - name: getty
-    image: "linuxkit/getty:e36b5e794256c2a56c000035f73a4d46f7ffb3e6"
```

Comment:
> pkg/getty/README.md is in pkg/getty and is therefore part of the hash, that's the problem I was getting at
> although in your proposal Makefile and Dockerfile would no longer be part of the hash but they should be because they impact the output

Maybe. I would tend more towards accepting the `Dockerfile` in there than the `Makefile`.

Truth is, I never really understood why we hash it this way. @justincormack or @rn explain it?

I think that the hash-based refs make the most sense: `linuxkit/getty@sha256:ac652...`. If you want it immutable, make it the generated contents of the image, which just works.